app/controllers/application_controller/ci_processing.rb

```diff
@@ -410,7 +422,13 @@ def process_objects(objs, task, display_name = nil)
     options = {:ids => objs, :task => task, :userid => session[:userid]}
     options[:snap_selected] = session[:snap_selected] if task == "remove_snapshot" || task == "revert_to_snapshot"
 
-    klass.process_tasks(options)
+    if task == 'retire_now'
+      options = {:task => task, :userid => session[:userid], :ids => objs, :src_ids => objs}
+      VmRetireRequest.make_request(@request_id, options, current_user)
```

Comment:
In the call this is modeled after, it sets the request_id to nil right before it makes this call (on the migrate side). Maybe I should be doing so too?